File: Data Structures & Algorithms/valid-tree/submission-0.go
func validTree(n int, edges [][]int) bool {
    s := make([]int, 0)
    if len(edges) != n-1 {
        return false
    }
    s = append(s, 0)
    g := make([][]int, n)
    for i := 0;i<n;i+=1{
        g[i] = make([]int, 0)
    }
    for _, e := range edges {
        u,v := e[0], e[1]
        g[u] = append(g[u],v)
        g[v] = append(g[v], u)
    }
    visited := make([]bool, n)

    
    for len(s) > 0 {
        f := s[0]
        s = s[1:]
        visited[f] = true
        for _, v := range g[f]{
            if !visited[v] {
                s = append(s, v)
            }
        }
    }
    for i:=0;i<n;i+=1{
        if !visited[i]{
            return false
        }
    }
    return true
}

File: Data Structures & Algorithms/min-cost-to-connect-points/submission-0.py
class Solution:
    def minCostConnectPoints(self, points: List[List[int]]) -> int:
        n = len(points)
        q = [(0,0)]
        visited = [False]*n
        cost = 0
        while len(q) > 0:
            front = heapq.heappop(q)
            if visited[front[1]]:
                continue
            cost += front[0]
            visited[front[1]] = True
            for i in range(0,n):
                if not visited[i]:
                    dist = abs(points[i][0]-points[front[1]][0]) + abs(points[i][1]-points[front[1]][1])
                    heapq.heappush(q, (dist,i))
        return cost
